packages/astro/src/assets/build/generate.ts

```diff
@@ -155,12 +155,12 @@ export async function generateImage(
 	};
 }
 
-export function getStaticImageList(): Iterable<
-	[string, { path: string; options: ImageTransform }]
+export function getStaticImageList(): Map<
+	string, { path: string; options: ImageTransform }
 > {
 	if (!globalThis?.astroAsset?.staticImages) {
-		return [];
+		return new Map();
```

Comment:
Returning a map forces me to create one here, which I presume isn't the worst, it's just once for an empty map.
The rest reduces a lot of overhead.

Reply:
That's fine with me!